lib/modules/manager/cargo/artifacts.spec.ts

```diff
@@ -112,6 +112,35 @@ describe('modules/manager/cargo/artifacts', () => {
     expect(execSnapshots).toMatchSnapshot();
   });
 
+  it('passes customEnvVariables through while updating the cargo lock', async () => {
+    GlobalConfig.set({
+      customEnvVariables: {
+        FOO_ENV: 'bar',
+      },
+    });
+
+    fs.statLocalFile.mockResolvedValueOnce({ name: 'Cargo.lock' } as any);
+    fs.findLocalSiblingOrParent.mockResolvedValueOnce('Cargo.lock');
+    git.getFile.mockResolvedValueOnce('Old Cargo.lock');
+    const execSnapshots = mockExecAll();
+    fs.findLocalSiblingOrParent.mockResolvedValueOnce('Cargo.lock');
+    fs.readLocalFile.mockResolvedValueOnce('New Cargo.lock');
+    const updatedDeps = [
+      {
+        depName: 'dep1',
+      },
+    ];
+    expect(
+      await cargo.updateArtifacts({
+        packageFileName: 'Cargo.toml',
+        updatedDeps,
+        newPackageFileContent: '{}',
+        config,
+      })
+    ).not.toBeNull();
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
no snapshots, checkout other managers for better matching

Reply:
Sorry - I was following the pattern in this test.
I'll have a look through.